src/generate_bindings.nim

```diff
@@ -0,0 +1,673 @@
+          # raymath header file contains implementation and c2nim crashes
+          # trying to parse it. Here implementation is stripped away and only
+          # definition is left.
+          rs.add line & ";\n"
+          if i+1 < raylibhLines.len and raylibhLines[i+1] == "{":
+            while raylibhLines[i] != "}": i.inc
+        elif words.len > 1 and words[0] == "#define" and words[1] in ignoreDefines:
+          echo "Ignore: " & line
+        else:
+          rs.add line & "\n"
+        i.inc
+
+      if nimColorNames.len > 0 and filename == "raylib":
+        rs.add fmt"""
+#ifdef C2NIM
+#@
+{nimColorNames}
+@#
+#endif
+"""
+      writeFile(buildDir/fmt"{filename}_modified.h", rs)
+
+
+    # Processing with c2nim
+
+    echo "\nExecuting c2nim"
+    let c2nimcmd = findExe("c2nim") & " " & buildDir/fmt"{filename}_modified.h"
+    echo c2nimcmd & "\n"
+    assert execCmd(c2nimcmd) == 0
+
+
+    # Postprocessing of generated nim file by c2nim
+
+    block postprocessing:
+      const
+        raymathShortcuts = slurp(templatesDir/"raymath_shortcuts.nim")
+        # proc beginTextureMode*(target: RenderTexture2D) {.cdecl,
+        #     importc: "BeginTextureMode", header: raylibHeader.}
+        # ##  Initializes render texture for drawing
+        reBeginProc = re"(?sm)^proc ((begin[^*]*)\*\(.*?\)).*"
+        # beginScissorMode*(x: cint; y: cint; width: cint; height: cint)
+        reArgumentType = re": [[:word:]]+;"
+        reArgumentTypeEnd = re": [[:word:]]+\)"
+        # proc vector2Zero*(
+        reMathTypeProc = re"^proc ((vector[23]|matrix|quaternion)[^*]+)\*.*"
+
+        # These cannot be determined automatically, only through manual inspection.
+        uncheckedArrayReplacements = [
+          # In Font
+          ("recs* {.importc: \"recs\".}: ptr Rectangle",
+           "recs* {.importc: \"recs\".}: ptr UncheckedArray[Rectangle]"),
+          ("glyphs* {.importc: \"glyphs\".}: ptr GlyphInfo",
+           "glyphs* {.importc: \"glyphs\".}: ptr UncheckedArray[GlyphInfo]"),
+          # In Shader
+          ("locs* {.importc: \"locs\".}: ptr cint",
+           "locs* {.importc: \"locs\".}: ptr UncheckedArray[cint]"),
+          # In Material
+          ("maps* {.importc: \"maps\".}: ptr MaterialMap",
+           "maps* {.importc: \"maps\".}: ptr UncheckedArray[MaterialMap]"),
+          # In Model
+          ("meshes* {.importc: \"meshes\".}: ptr Mesh",
+           "meshes* {.importc: \"meshes\".}: ptr UncheckedArray[Mesh]"),
+          ("materials* {.importc: \"materials\".}: ptr Material",
+           "materials* {.importc: \"materials\".}: ptr UncheckedArray[Material]"),
+          ("meshMaterial* {.importc: \"meshMaterial\".}: ptr cint",
+           "meshMaterial* {.importc: \"meshMaterial\".}: ptr UncheckedArray[cint]"),
+          # In ModelAnimation
+          ("framePoses* {.importc: \"framePoses\".}: ptr ptr Transform",
+           "framePoses* {.importc: \"framePoses\".}: ptr UncheckedArray[ptr Transform]"),
```

Comment:
That's a ptr UncheckedArray[ptr UncheckedArray[Transform]]

Reply:
Good catch!